### PR TITLE
Enabled disabled tests for netcoreapp in System.Diagnostics.Tracing and updated missing ref members

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/corefx/issues/4865

### CODE REVIEW

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/Listeners.cs [hunk 1]
```diff
@@ -348,82 +346,76 @@ public override void EventSourceCommand(string eventSourceName, EventCommand com
...
-                if (_forwardTo != null && _forwardTo._onEventSourceCreated != null)
-                    _forwardTo._onEventSourceCreated(eventSource);
+                _forwardTo?._onEventSourceCreated?.Invoke(eventSource);
             }
 
-            private EventListenerListener _forwardTo;
+            protected override void OnEventWritten(EventWrittenEventArgs eventData)
+            {
+#if netcoreapp
+                // EventSource:L4375 OnEventWritten is internal protected. Can't call...
```
Comment (ViktorHofer) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/Harness/Listeners.cs]: this line is definitely problematic... Not sure how to solve it with the current architecture (internal protected vs packaged)...

**Comment by ViktorHofer** [comment]
ping @vancem @brianrob

**Comment by karelz** [comment]
@vancem can you please comment? or @brianrob?

**Comment by joperezr** [comment]
One thing we have been trying to do in cases like this are to try to separate these tests into netcoreapp specific files that only get includded in the netcoreapp configuration, so have the tests be partial classes and create files `TestsWriteEventToListener.netcoreapp.cs` and `Listeners.netcoreapp.cs` and only add those when `'$(TargetGroup)' == 'netcoreapp'`. However, if we feel that we will eventually run this tests in more platforms, then #ifdefing pattern is fine.

**Comment by joperezr** [comment]
other than that small comment, at least the configuration part looks good to me. seems like CI builds are too old so they got cleaned and we can't look at the failing logs, @ViktorHofer did you get a chance to look into them and try to see what was failing?

**Comment by ViktorHofer** [comment]
@dotnet-bot test this please

**Comment by ViktorHofer** [comment]
@joperezr restarted CI, can't remember the error messages. But the tests are also failing locally when you run msbuild. If you have some cycles, you could clone my fork and test it to see what exactly is failing. Maybe you'll find some parts which need to be updated.

**Comment by danmosemsft** [comment]
@joperezr could you maybe try this locally to unblock @ViktorHofer ?

**Comment by ViktorHofer** [comment]
Had finally time to finish my PR. @vancem PTAL

**Comment by ViktorHofer** [comment]
@joperezr I would keep the ifdef for now (instead of own classes) as with netfx >= 462 (and therefore I guess netstandard21) all tests are runnable and executable. Therefore at that point of time (soon) we will just need to remove the ifdefs.

**Comment by joperezr** [comment]
Sounds good to me

**Comment by ViktorHofer** [comment]
ping @vancem @brianrob

**Comment by vancem** [comment]
LGTM

**Pull request merged**